ui/main.qml

```diff
@@ -45,6 +45,18 @@ StatusWindow {
     }
     visible: true
 
+    onWidthChanged: windowResizeTimer.start()
+    onHeightChanged: windowResizeTimer.start()
+
+    Timer {
+        id: windowResizeTimer
+        interval: 3000
```

Comment:
This is just some arbitrary value. We might want to tweak this.

Essentially what this does is, it throttles the resize event so we don't write to disk (settings) on for every pixel when resized.

Reply:
@PascalPrecht timers are generally slow and usually having a lot of timers may have a performance impact. So please remove this timer and try using `Qt.callLater()` instead, since this is a perfect case for using it.

```
function storeWith(){
  localAppSettings.appWidth = width
}

function storeHeight(){
  localAppSettings.appHeight = height
}

onWidthChanged: Qt.callLater(storeWith)
onHeightChanged: Qt.callLater(storeHeight)
```
This should reduce number of calls, and also store only property which has been changed.

Reply:
Thanks for the suggestion!

> This should reduce number of calls, and also store only property which has been changed.

This doesn't seem to be the case. I'm observing this to behave the same as if there wasn't a `Qt.callLater` at all.

One of the reasons I've introduced the timer, is so that we **don't** write `appWidth` and `appHeight` on every pixel that triggers the width/heightChanged events (because it eventually causes the app to write to disk and we want to keep that at a minimum).

There doesn't seem too be a different in call count when using Qt.callLater.

Is there something I'm missing, or anything I can do to inspect this further?

Reply:
@PascalPrecht when I said "This should reduce number of calls", I mostly meant if user resizes the app's width we don't need to store height since it will remain the same and vice versa (one write to the file (one property) instead of two, but that shouldn't be a big deal, just. wanted to mention).

Regarding a `Qt.callLater`, [here](https://doc.qt.io/qt-5/qml-qtqml-qt.html#callLater-method-1) is a description, so if we are going to trust Qt documentation, it says "When this function is called multiple times in quick succession with the same function as its first argument, that function will be called only once" (there is an example), now a question is what is "quick sessions" is that every 10ms or 100ms or what.

> There doesn't seem too be a different in call count when using Qt.callLater.

Maybe, didn't do a measurements, just said what is in the docs and that this is a perfect place for using such function.

Even if you decide to keep Timer, reduce interval to 200-500ms, or just use `Backpressure.debounce` we have on the qml side, but it also creates a timer which would be good to avoid as much as we can.

Reply:
> @PascalPrecht when I said "This should reduce number of calls", I mostly meant if user resizes the app's width we don't need to store height since it will remain the same and vice versa (one write to the file (one property) instead of two, but that shouldn't be a big deal, just. wanted to mention).

Oh I see, understood!

> now a question is what is "quick sessions" is that every 10ms or 100ms or what.

Yea, this is kinda important.
I've only put some logs in the functions to be called and they seems to trigger as often as without going through `callLater` (to be fair: I didn't count the calls, but it's clearly much more than waiting a few ms until we write to disk). This is probably the case because Qt is fast enough to return to the event loop, so it happily schedules the next task and just goes on with it.

I'm all in favour of using native APIs though, so let's go with `callLater` for now and see if that causes issues when people use this feature.

> timers are generally slow and usually having a lot of timers may have a performance impact. 

Just out of curiosity: Isn't `Timer` leveraging the same event loop as `Qt.callLater` ? Since `Timer` also just does  a `setTimeout`, I'm wondering how that is different from scheduling a task via `callLater`... (both put the task in a queue to be executed once the stack is done)

Any insights on that would be great! :D

Reply:
@PascalPrecht correct we cannot use timers if we don't have an event loop. 
In short about the difference Qt.callLater should discard the same function calls between two returnings from the event loop, when it calls that function. But timer may be a hard component for the app, depends on accuracy, and that may varies depends on OS and how complex UI and how many updates on the UI we do meanwhile. This app at this moment is not too demanding, but in case we have some realtime chart later for example to display realtime bid/ask for all cryptos user have in the app, then using timers without control and good management may impact app performances.

All in all, as far as I've had a chance to see the best options for timers, of course if you don't need high acurency, were from QObject, since it provides basic timer support in Qt with minimal overhead.

For this app I have some idea to introduce a global timer on the app level and implement an observer pattern for that, in that case we may use the same timer for everything, but we may discuss that later, still not mandatory.